dataset/src/main/spark-32/frameless/FoldableImpl.scala

```diff
@@ -0,0 +1,54 @@
+package frameless
+
+import org.apache.spark.sql.FramelessSpark32Internals
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.{CaseWhen, Coalesce, Expression, If, NaNvl, UnaryExpression}
+import org.apache.spark.sql.catalyst.expressions.objects.InvokeLike
+import org.apache.spark.sql.types.{DataType, ObjectType}
+
+object FoldableImpl {
```

Comment:
Are you sure we need it? Why not to add an alias into `FramelessInternals`? It won't fix the Spark 3.2 issue, but will make it compile.

Reply:
Probably not.  I'd guess it's not likely there are huge Lit expression trees that need transformUp'ing that it's worth saving the computational time.

The only risk of doing that is they change the impl in 3.3/3.4 so it's no longer correct for them, and that's probably only for InovkeLike, I can't imagine an need to change the ConditionalExpression logic.

Reply:
I'd say lets add the less invasive change, don't address possible Spark 3.2 problems, and we'll get rid of 3.2 with the 3.5 release.

Reply:
sorry, it's been a day without much coffee, I don't understand that last bit, it seems contradictory. I seriously need more coffee but I'd like to sleep - ah the coders dilemma.

do you want either:

a) 3.2 issues are addressed by moving spark-32/frameless/FoldableImpl into FramelessInternals fully and accepting that this will be run on 3.3 and 3.4 with a possible risk of divergent implementation. < the pushdown tests run all versions and the project structure stays as per the previous release / #699 
b) we don't address 3.2 issues and so we don't need FramelessInternals as we can just call catalystExpr.foldable  < the tests should be disabled on 3.2 (but with project structure as per #699), or
c) yet something else?

Reply:
1. setting `foldable = true for Spark 3.2`, invoke `catalystExpr.foldable` for 3.3+
2. setting `foldable = false for Spark 3.2`, invoke `catalystExpr.foldable` for 3.3+, but put the test into the Spark 3.3+ test folder, similar to what we've done with `MapGroups`.

Reply:
ok, 1 has the risk of failure per [SPARK-40380](https://issues.apache.org/jira/browse/SPARK-40380) so I can't recommend that.

I'd offer a 3 in that case.:

use catalystExpr.foldable (equates to false on 3.2 but is fully compatible behaviour wise) and split the tests.

Reply:
Yea, makes sense :+1: So later on we just remove the tests, awesome.